cli/launchpad/runtime.go

```diff
@@ -0,0 +1,154 @@
+// Package launchpad file runtime.go contains runtime related support for
+// evaluation hierarchy and evaluated object tracking.
+//
+// In the global state, a evaluation stack is needed to support YAML evaluation. During evaluation, a
+// depth-first search approach is employed. For example,
+//
+//   kind: Folder
+//   spec:
+//     id: X
+//     folders:
+//       - id: Y
+//
+// folder Y will be evaluated first, then folder X. In this case, a stack can help infer parent's attribute
+// during the evaluation of folder Y.
+package launchpad
+
+import (
+	"errors"
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"log"
+)
+
+var (
+	errDuplicatedDefinition = errors.New("duplicated definition")
+	errOrgIdConflict        = errors.New("conflicted organization definition")
+	errUndefinedReference   = errors.New("undefined references")
+)
+
+// globalState stores metadata such as stack and evaluated objects to facilitate output generation.
+type globalState struct {
+	stack           []stackFrame
+	outputDirectory string
+	outputFlavor    outputFlavor
+	references      map[string]*parentRefYAML
+	evaluated       struct {
+		orgId   string
+		folders folders
+	}
+}
+
+// ==== Evaluation Stack ====
+
+// stackable interface determines what can be pushed onto the stack.
```

Comment:
This is *not* defining an interface. This is just creating a new type that is the same as an existing type.

A stackable interface might look something like:

```go
type stackable interface {
  Unmarshaler([]byte) error
}
```

That means that every object that implements the `Unmarshaler` method described will implicitly implement the `stackable` interface.

What you have here is *not* an interface in any way and is just over-complicating the code with the illusion of making it more flexible.